### PR TITLE
Set project variable to allow for multi-project deployments

### PR DESCRIPTION
**Pull request by GNewbury1**
# What this PR does
This MR allows specifying of which project `vault-cluster` will deploy to, by adding a project variable for each vault instance as `${var.gcp_project_id}`, as well as a project variable for the network and firewall rules, `${var.network_project_id}`.

# Why this PR is needed
The current method uses the project defined in the 'provider' block. This is good for single-project deployments, but when using `vault-cluster` with other terraform files which need deploying into other projects, there will be no project specified in `provider`. Each terraform object will have it's project specified instread. This PR will make sure that `vault-cluster` has the same ability.

### CODE REVIEW

**Comment by josh-padnick** [comment]
LGTM, but again we need to update the `main.tf` in the root folder to include `gcp_project_id`. Then ready to merge!

**Comment by GNewbury1** [comment]
That should now be done. Also formatted `main.tf` with `terraform fmt`.

**Comment by GNewbury1** [comment]
I've now also formatted the `main.tf` of `vault-cluster`

File: main.tf [hunk 1]
```diff
@@ -25,23 +24,25 @@ module "vault_cluster" {
   # source = "git::git@github.com:hashicorp/terraform-google-vault.git//modules/vault-cluster?ref=v0.0.1"
   source = "modules/vault-cluster"
 
+  project = "${var.gcp_project}"
```
Comment (josh-padnick) [inline main.tf]: Should this be:

```hcl
gcp_project_id = "${var.gcp_project_id}" 
```

Note that we should use the same var name in both examples and modules for consistency

File: modules/vault-cluster/main.tf [hunk 2]
```diff
@@ -64,8 +67,8 @@ resource "google_compute_instance_template" "vault_public" {
   }
 
   network_interface {
-    network = "${var.subnetwork_name != "" ? "" : var.network_name}"
-    subnetwork = "${var.subnetwork_name != "" ? var.subnetwork_name : ""}"
+    network = "${var.network_name}"
```
Comment (josh-padnick) [inline modules/vault-cluster/main.tf]: Why revert these changes?

**Comment by GNewbury1** [comment]
Just added a fairly major change - the specifying the project for the network is now different to specifying the project for the compute instances. This allows for deployments which may use a network shared from a different project.

**Comment by josh-padnick** [comment]
Ok, updates made. I'll merge this now but will hold off on issuing a release until we have automated tests that officially validate these changes.
